src/components/utils/popover.ts

```diff
@@ -381,30 +480,92 @@ export default class Popover extends EventsDispatcher<PopoverEvent> {
       }));
     }
 
+    if (item.isActive) {
+      el.classList.add(Popover.CSS.itemActive);
+    }
+
     return el;
   }
 
   /**
    * Item click handler
    *
    * @param itemEl - clicked item
+   * @param event - click event
    */
-  private itemClicked(itemEl: HTMLElement): void {
+  private itemClicked(itemEl: HTMLElement, event: MouseEvent): void {
     const allItems = this.nodes.wrapper.querySelectorAll(`.${Popover.CSS.item}`);
     const itemIndex = Array.from(allItems).indexOf(itemEl);
     const clickedItem = this.items[itemIndex];
 
-    clickedItem.onClick(clickedItem);
+    this.cleanUpConfirmationState();
+
+    if (clickedItem.confirmation) {
+      /** Save root item requiring confirmation to restore original state on popover hide */
+      if (this.itemRequiringConfirmation === null) {
+        this.itemRequiringConfirmation = clickedItem;
+      }
+      this.items[itemIndex] = clickedItem.confirmation;
+      const confirmationStateItemEl = this.createItem(clickedItem.confirmation as PopoverItem);
+
+      confirmationStateItemEl.classList.add(Popover.CSS.itemConfirmation);
+      itemEl.parentElement.replaceChild(confirmationStateItemEl, itemEl);
+
+      this.reactivateFlipper(
+        this.flippableElements,
+        this.flippableElements.indexOf(confirmationStateItemEl)
+      );
+
+      return;
+    }
```

Comment:
maybe it would be better to move this code to the `enableConfirmationStateForItem(item)` ?

Reply:
yep, moved the code to separate method